File: src/lib/types/k8s/services.ts
// Service-related types
export interface K8sService {
  name: string;
  namespace: string;
  type_: string;
  clusterIp: string;
  externalIp: string;
  ports: string;
  age: string;
}

export interface ServiceFilters {
  labels?: string[];
  fieldSelectors?: string[];
}

File: src-tauri/src/k8s/types/services.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct K8sService {
    pub name: String,
    pub namespace: String,
    pub type_: String,
    pub cluster_ip: String,
    pub external_ip: String,
    pub ports: String,
    pub age: String,
}
